### DISCUSSION TITLE
重新render如何保持滚动条位置

### DISCUSSION THREAD

**Discussion by KuduroJS** [antvis/G2#5655] [Q&A] [answered]
https://github.com/antvis/G2/assets/101621239/3867d3bd-4127-4449-803e-5c6ef2c1b9c9

需求如上，联动过滤。

现在的问题是，移动滚动条之后，联动重新渲染会重置滚动条的位置，用户体验一般。我希望滚动条保持在那个位置。

我的想法是记录一下滚动条位置，重新渲染的时候，移动滚动条。

请问，如何记录滚动条位置呢？有相关事件吗？

或是其他任何建议？谢谢。

**Comment by pearmini** [comment, marked as answer]
需要自己通过 `chart.on('sliderX:filter', () => {})` 记录状态，然后通过 `chart.emit('sliderX:filter`, ...)` 去恢复状态
